lib/rules/no-sync.js

```diff
@@ -32,7 +39,65 @@ module.exports = {
                     },
                     ignores: {
                         type: "array",
-                        items: { type: "string" },
+                        items: {
+                            oneOf: [
+                                { type: "string" },
+                                {
+                                    type: "object",
+                                    properties: {
+                                        from: {
+                                            type: "string",
+                                            enum: ["file"],
+                                        },
```

Comment:
Maybe const is better here?

```suggestion
                                        from: { const: "file" },
```